thinc/extra/wrappers.py

```diff
@@ -93,3 +94,48 @@ def use_params(self, params): # pragma: no cover
         yield
         self.from_bytes(backup)
 
+class PyTorchWrapperRNN(PyTorchWrapper):
+    '''Wrap a PyTorch RNN model
+    '''
+    def begin_update(self, x_data, h_0=None, drop=0., sgd=None):
+        '''Return the output of the wrapped PyTorch model for the given input,
+        along with a callback to handle the backward pass.
+        '''
+        x_var = torch.autograd.Variable(torch.Tensor(x_data),
+                                        requires_grad=True)
+        # Make prediction
+        out, h_n = self._model(x_var, h_0)
+        # Shapes will be:
+        # out = seq_len, batch, hidden_size * num_directions
+        # h_n = num_layers * num_directions, batch, hidden_size
+
+        def backward_pytorch(dy_data, sgd=None):
+            dout = torch.autograd.Variable(torch.Tensor(dy_data))
+            torch.autograd.backward((out,), grad_variables=(dout,))
+            dX = self.ops.asarray(x_var.grad.data)
+            if sgd is not None:
+                if self._optimizer is None:
+                    self._optimizer = self._create_optimizer(sgd)
+                self._optimizer.step()
+                self._optimizer.zero_grad()
+            return dX
+        # TODO: We should return also h_n e.g. for seq2seq models
```

Comment:
Sure, it's probably a good thing to return a tuple. But then we should also change the gradient argument to `backward_pytorch_rnn` as well: the output from the forward pass and the argument to the backward pass should always line up (as should the argument to the forward pass and the output of the backward pass).

Reply:
Returning a tuple isn't so inconvenient, because the model can be wrapped in the `with_getitem` adapter from `thinc.api`. For instance, if you want the first element of the tuple (and want to send back a gradient of that element), you'd do `with_getitem(0, PytorchWrapperRNN())`